packages/office-ui-fabric-react/src/common/shallowUntilTarget.ts

```diff
@@ -0,0 +1,64 @@
+import { shallow, ShallowWrapper } from 'enzyme';
+
+/**
+ * Duplicated enzyme's ShallowRendererProps
+ *
+ * @internal
+ */
+export interface IShallowRendererProps {
+  lifecycleExperimental?: boolean;
+  disableLifecycleMethods?: boolean;
+}
+
+/**
+ * ShallowUntilTarget Interface
+ *
+ * @internal
+ */
+export interface IShallowUntilTarget {
+  maxTries: number;
+  shallowOptions: IShallowRendererProps;
+}
+
+/**
+ * An extention of enzyme's shallow function which will fail to work
+ * with decorated components and/or components using the styled() function.
+ * This function allows you to pass a 'target' component (e.g. ComponentBase)
+ * and keep running shallow on each child component till a match is found.
+ *
+ * @public
+ */
+export function shallowUntilTarget<P, S>(
+  componentInstance: React.ReactElement<P>,
+  TargetComponent: string,
```

Comment:
You could have target component be type of `Function`, and then users could pass in the actual class that they care about. You could then access the string by calling `TargetComponent.name` and it would work out fine. That way it keeps things a bit more type safe since they would need to import the actual class they care about. This is just a suggestion, if you feel like trying it out that would be cool, otherwise no worries.

Reply:
This was the original way that I had implemented this, but specifically for decorated functions like ResizeGroupBase, the name becomes 'ComponentWithInjectedProps' so it returns before it should.  Passing a string name into the function was a way to avoid this.